l1-contracts/contracts/bridge/L1ERC20Bridge.sol

```diff
@@ -167,14 +173,18 @@ contract L1ERC20Bridge is IL1ERC20Bridge, ReentrancyGuard {
         }
         // We don't need to set finalizeWithdrawal here, as we set it in the shared bridge
 
-        (address l1Receiver, address l1Token, uint256 amount) = SHARED_BRIDGE.finalizeWithdrawalLegacyErc20Bridge({
-            _l2BatchNumber: _l2BatchNumber,
-            _l2MessageIndex: _l2MessageIndex,
-            _l2TxNumberInBatch: _l2TxNumberInBatch,
-            _message: _message,
-            _merkleProof: _merkleProof
+        FinalizeL1DepositParams memory finalizeWithdrawalParams = FinalizeL1DepositParams({
+            chainId: ERA_CHAIN_ID,
+            l2BatchNumber: _l2BatchNumber,
+            l2MessageIndex: _l2MessageIndex,
+            l2Sender: L2_ASSET_ROUTER_ADDR,
```

Comment:
in the past, the sender was the address of the l2 shared bridge.

however, in this method we always use `L2_ASSET_ROUTER_ADDR`. It seems like there is no way people can finalize old (old means right before the l2 asset router upgrade) withdrawals